scripts/circleci/release/installScalyrAgentV2.sh

```diff
@@ -405,24 +430,51 @@ else
-  run_command "apt-get -y install gnupg"
-  # Now install the repo package, which must be done using `dpkg` since it is a package on local disk.
-  # Do not use `apt-get install` here. Even though it does now support installing a package from local disk, older
-  # versions run by our customers do not.
-  run_command "dpkg -i $TMPDIR/*.deb";
-
-  # We need an apt-get update to force apt to refresh the list of repositories
-  # and available packages.  Since this will cause all to refresh,
-  # it takes a little bit of time, but we do not have a choice.
-  # Also, we specifically check for an error some distros experience where apt-transport-https is not installed to
-  # give a better error message.  We cannot depend on that package because not all separate out the https on its own.
-  lecho "  Refreshing available package list (may take several minutes)";
-  run_command "apt-get update" check_for_https_error;
-
-  lecho "  Updating Scalyr yum repository configuration files using repository.";
-  run_command "apt-get -y install scalyr-repo";
+  if $USE_BOOTSTRAP_PACKAGES; then
+    echo "  Installing Scalyr apt repository configuration files.";
+    # Manually install dependencies for the repo package, since `dpkg -i` doesn't install them for you.
+    run_command "apt-get -y install gnupg"
+    # Now install the repo package, which must be done using `dpkg` since it is a package on local disk.
+    # Do not use `apt-get install` here. Even though it does now support installing a package from local disk, older
+    # versions run by our customers do not.
+    run_command "dpkg -i $TMPDIR/*.deb";
+
+    # We need an apt-get update to force apt to refresh the list of repositories
+    # and available packages.  Since this will cause all to refresh,
+    # it takes a little bit of time, but we do not have a choice.
+    # Also, we specifically check for an error some distros experience where apt-transport-https is not installed to
+    # give a better error message.  We cannot depend on that package because not all separate out the https on its own.
+    lecho "  Refreshing available package list (may take several minutes)";
+    run_command "apt-get update" check_for_https_error;
+
+    lecho "  Updating Scalyr yum repository configuration files using repository.";
+    run_command "apt-get -y install scalyr-repo";
+  else
+    # update repository and install tools that may be needed.
+    run_command "apt-get -y update";
+
+    echo "Install required dependencies."
+    export DEBIAN_FRONTEND=noninteractive
+    run_command "apt-get install -y gnupg"
+
+
+    apt_version=$(dpkg -s apt | grep -i version | awk '{print $2}')
+
+    # the apt-key tool is deprecated since debian 10 and is removed in debian 11,
+    # so we just manipulate key files directly through filesystem on newer versions.
+    # see https://manpages.debian.org/testing/apt/apt-key.8.en.html#DESCRIPTION
+    if [[ "$apt_version" < "1.8.2.1" ]]; then
```

Comment:
Is there any benefit to use using the `apt-key` tool for the older versions?  Could we just use the same approach (modify the files directly) in both cases?  It seems to me, having different implementations for the different versions is just one more thing that can go wrong.

Reply:
Yeah, Now I think so too. Done.